hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/MintPrecompile.java

```diff
@@ -98,10 +94,14 @@ public MintPrecompile(
     @Override
     public TransactionBody.Builder body(final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
         this.transactionBody = null;
-        mintOp = switch (functionId) {
-            case AbiConstants.ABI_ID_MINT_TOKEN -> decodeMint(input);
-            case AbiConstants.ABI_ID_MINT_TOKEN_V2 -> decodeMintV2(input);
-            default -> null;};
+        final var mintAbi =
+                switch (functionId) {
+                    case AbiConstants.ABI_ID_MINT_TOKEN -> SystemContractAbis.MINT_TOKEN_V1;
+                    case AbiConstants.ABI_ID_MINT_TOKEN_V2 -> SystemContractAbis.MINT_TOKEN_V2;
+                    default -> throw new IllegalArgumentException("invalid selector to mint precompile");
+                };
+        mintOp = getMintWrapper(input, mintAbi);
+        if (mintOp == null) throw new IllegalArgumentException("unable to create mint wrapper from decoded input");
```

Comment:
Same question for null `MintWrapper` cases as noted in the burn case

Reply:
same answer ...

Reply:
Okay thanks